packages/blockchain/src/index.ts

```diff
@@ -267,12 +213,30 @@ export default class Blockchain implements BlockchainInterface {
     this.db = opts.db ? opts.db : level()
     this.dbManager = new DBManager(this.db, this._common)
 
+    if (opts.consensus) {
+      this.consensus = new opts.consensus(
+        this.db,
+        this.dbManager,
+        this._common,
+        this._validateConsensus
+      )
+    } else if (this._common.consensusAlgorithm() === ConsensusAlgorithm.Clique) {
```

Comment:
To bring this together with Ryans question on POS consensus: yes there should also be a consensus class `POS`, this is our third used consensus type, if I am not forgetting anything the initialization should throw in all other cases.

This would then used here instead of initializing on this "base" consensus, implementation should be pretty simple, in most cases there will just be no or very little functionality within the function calls (which is ok).

Reply:
I added a new class called `CasperConsensus` since the other two (`EthashConsensus` and `CliqueConsensus`) corresponded with the consensus algorithm, but let me know if you think it should be called `POSConsensus`